Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -731,21 +760,27 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
                      collisionTimeMask);
 
     auto trackRef = primVer2TRefs[collisionID];
-    int start = trackRef.getFirstEntryOfSource(0);
-    int ntracks = trackRef.getEntriesOfSource(0);
-    // FIXME: `track<-->vertex` ambiguity is not accounted for in this code
-    for (int ti = 0; ti < ntracks; ti++) {
-      auto trackIndex = primVerGIs[start + ti];
-      const auto source = trackIndex.getSource();
-      // setting collisionID for tracks attached to vertices
-      if (source == GIndex::Source::TPC) {
-        vCollRefsTPC[trackIndex.getIndex()] = collisionID;
-      } else if (source == GIndex::Source::ITS) {
-        vCollRefsITS[trackIndex.getIndex()] = collisionID;
-      } else if (source == GIndex::Source::ITSTPC) {
-        vCollRefsTPCITS[trackIndex.getIndex()] = collisionID;
-      } else {
-        LOG(WARNING) << "Unsupported track type!";
+    for (auto src : {GIndex::Source::ITS, GIndex::Source::TPC, GIndex::Source::MFT, GIndex::Source::ITSTPC}) {
+      int start = trackRef.getFirstEntryOfSource(src);
+      int ntracks = trackRef.getEntriesOfSource(src);
+      LOG(DEBUG) << " ====> Collision " << collisionID << " ; src = " << src << " : ntracks = " << ntracks;
+
+      // FIXME: `track<-->vertex` ambiguity is not accounted for in this code
+      for (int ti = 0; ti < ntracks; ti++) {
+        auto trackIndex = primVerGIs[start + ti];
+        const auto source = trackIndex.getSource();
```

Comment:
Hi Ruben, source has been replaced by SRC. The loop remains on the supported types to avoid flooding the output with the unsupported track type warning.